### PR TITLE
Added: Added game setting to enable verbose logging

### PR DESCRIPTION
**Pull request by jwvanderbeck**
Added a new setting for verbose logging in the game settings.  Defaults to FALSE.

Wrapped calls to ROLLog.log and ROLLog.debug up in a check against the user's verbose log setting and only do the logging if set.

Logging has a performance hit, and even preparing a string to log can cause boxing and allocations.  This small cleanup won't make a huge difference with so many mods but its a start and the right way to do it :)

### CODE REVIEW

File: Source/ROLib/Utils/ROLGameSettings.cs [hunk 1]
```diff
@@ -8,6 +8,8 @@ public class ROLGameSettings : GameParameters.CustomParameterNode
         [GameParameters.CustomParameterUI("Persistent Recolor Selections", toolTip = "If true, custom recolor selections will persist across texture set changes.")]
         public bool persistRecolorSelections = false;
 
+        [GameParameters.CustomParameterUI("Enable Verbose Llogging", toolTip = "Additional Verbose logging can help when troubleshooting, but also reduces performance.")]
+        public bool enabledDebugLogging = false;
```
Comment (siimav) [inline Source/ROLib/Utils/ROLGameSettings.cs:12]: LLogging? Also needs a newline there.

**Comment by DRVeyl** [comment]
I disagree with the implementation.  It is not practical to guard every log call with an if statement.  If anything, this should be placed into the ROLLog utility -- consider changing the static debug enabling variable there to reference your new GameParameter so it can be changed at runtime.  That I'd support.

Yes, you will absolutely eat the cost of the string allocation, which is essentially meaningless in the cases where these are getting called.  This matters if you are generating strings in Update().  Not in methods called in Start(), event-handlers, or otherwise single-shot components.

I don't believe the spamming in the Editor is performance-impacting, as it should not fire more than each time you click.  So I don't think we are at all talking performance, but more log readability.  Which points me back to... change ROLLog.

(Note I, at least, am not consistent in using the ROLLog tool.  You'll find normal UnityEngine.Debug.Log() calls, too.)

**Comment by jwvanderbeck** [comment]
Guarding every call adds code and makes it a bit bulky, but the performance hit is far less even than the actual `Debug.Log()` call and that doesn't even take into account boxing allocations.

The overall performance increase here is minor.  It is more a start to a goal to guide people onto a better path.  This is something that has been bothering me for a while now, and nearly every single mod, including TestFlight, is way too loose with logging.

As for Update versus other less used instances; yes the overall gain is far less important here but it is a matter of consistency for the most part.  Additionally, even in the case of "each time you click" there is overhead.  A flurry of log statements that fire, when you click (not that these appear to be a flurry), can still case a second of delay when click to pick up a part or swapping between models flor example and these all add up to a degradation of experience.

Moving it into the Log utility is a good idea, which I had thought of, but in order to make it actually useful, you have to then offer a collection of overloaded methods to build strings as well because the boxing allocations used to build string before logging is part of the problem.  And using those methods won't be as nice and clean as normal string concatenation.  Which in turn means people won't want to use them.

**Comment by DRVeyl** [comment]
Again, I disagree with trading readability for what I consider a negligible performance improvement, and the bulkiness caused from wrapping rarely-invoked debug strings is part of that trade.  In the specific cases where that performance trade is worth-while, ie many-calls-per-second paths such as Update/FixedUpdate, then a guard technique like this I agree is appropriate.  That is none of these commits.

I have been cutting out various unnecessary debug logging calls in the parts of the ROLibrary I've been touching.  Because I've had to also read through their output, and some portions of this remain very work-in-progress.

**Pull request closed without merging**